book-my-show/backend/java/bms-monolith/src/main/java/org/lbcc/bms/bms_monolith/vendor/controller/VendorOnboardingController.java

```diff
@@ -0,0 +1,41 @@
+package org.lbcc.bms.bms_monolith.vendor.controller;
+
+import org.lbcc.bms.bms_monolith.common.enums.VendorStatus;
+import org.lbcc.bms.bms_monolith.vendor.dto.VendorOnboardRequestDto;
+import org.lbcc.bms.bms_monolith.vendor.service.VendorService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RequestMapping("/vendor")
+@RestController
+public class VendorOnboardingController {
+
+    private final VendorService vendorService;
+
+    @Autowired
+    public VendorOnboardingController(VendorService vendorService) {
+        this.vendorService = vendorService;
+    }
+
+    //TODO: this will be allowed only by ADMIN
+    @PostMapping("/onboard")
+    public ResponseEntity<?> onboardNewVendor(@RequestBody VendorOnboardRequestDto vendorOnboardRequestDto) {
```

Comment:
use APIResponse used in the rest of the project, refer to EventController.

change name VendorOnboardRequestDto  to VendorOnboardRequest: Request itself is self-explanatory.